core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1655,6 +1663,217 @@ class KafkaApis(val requestChannel: RequestChannel,
+            new SecurityDisabledException("No Authorizer is configured on the broker."))
+        sendResponseMaybeThrottle(request, createResponse)
+      case Some(auth) =>
+        val errors = mutable.HashMap[Int, Throwable]()
+        var creations = ListBuffer[(Resource, Acl)]()
+        for (i <- 0 to createAclsRequest.aclCreations().size() - 1) {
+          val result = toScala(createAclsRequest.aclCreations.get(i).acl.toFilter)
+          result match {
+            case Failure(throwable) => errors.put(i, throwable)
+            case Success((resource, acl)) => try {
+                if (resource.resourceType.equals(Cluster) &&
+                    !resource.name.equals(Resource.ClusterResourceName))
+                  throw new InvalidRequestException("The only valid name for the CLUSTER resource is " +
+                      Resource.ClusterResourceName)
+                if (resource.name.isEmpty())
+                  throw new InvalidRequestException("Invalid empty resource name")
+                auth.addAcls(immutable.Set(acl), resource)
+              } catch {
+                case throwable : Throwable => errors.put(i, throwable)
+              }
+          }
+        }
+        var aclCreationResults = new java.util.ArrayList[AclCreationResponse]
+        for (i <- 0 to createAclsRequest.aclCreations().size() - 1) {
+          errors.get(i) match {
+            case Some(throwable) => aclCreationResults.add(new AclCreationResponse(throwable))
+            case None => aclCreationResults.add(new AclCreationResponse(null))
+          }
+        }
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          new CreateAclsResponse(throttleTimeMs, aclCreationResults)
+        sendResponseMaybeThrottle(request, createResponse)
+    }
+  }
+
+  def handleDeleteAcls(request: RequestChannel.Request): Unit = {
+    authorizeClusterAction(request)
+    val deleteAclsRequest = request.body[DeleteAclsRequest]
+    authorizer match {
+      case None =>
+        def createResponse(throttleTimeMs: Int): AbstractResponse =
+          deleteAclsRequest.getErrorResponse(throttleTimeMs,
+            new SecurityDisabledException("No Authorizer is configured on the broker."))
+        sendResponseMaybeThrottle(request, createResponse)
+      case Some(auth) =>
+        val filterResponseMap = mutable.HashMap[Int, AclFilterResponse]()
+        var toDelete = mutable.HashMap[Int, ListBuffer[(Resource, Acl)]]()
+        for (i <- 0 to deleteAclsRequest.filters().size - 1) {
+          toDelete.put(i, new ListBuffer[(Resource, Acl)]())
+        }
+        if (deleteAclsRequest.filters().asScala.exists { f => !f.matchesAtMostOne() }) {
+          // Delete based on filters that may match more than one ACL.
+          val aclMap : Map[Resource, Set[Acl]] = auth.getAcls()
+          aclMap.foreach {
+            case (resource, acls) => {
+              acls.foreach {
+                case (acl) => {
```

Comment:
There's no need for `case` here. Also, the parameter can be in the previous line and we don't need the `{` after `=>`.